packages/wallet/src/wallets.ts

```diff
@@ -46,30 +47,35 @@ export class WalletUnlocked extends BaseWalletUnlocked {
   /**
    * Create Wallet Unlocked from a seed
    */
-  static fromSeed(seed: string, path?: string): WalletUnlocked {
+  static fromSeed(seed: string, path?: string, provider?: Provider): WalletUnlocked {
     const hdWallet = HDWallet.fromSeed(seed);
     const childWallet = hdWallet.derivePath(path || WalletUnlocked.defaultPath);
 
-    return new WalletUnlocked(<string>childWallet.privateKey);
+    return new WalletUnlocked(<string>childWallet.privateKey, provider);
   }
 
   /**
    * Create Wallet Unlocked from mnemonic phrase
    */
-  static fromMnemonic(mnemonic: string, path?: string, passphrase?: BytesLike): WalletUnlocked {
+  static fromMnemonic(
+    mnemonic: string,
+    path?: string,
+    passphrase?: BytesLike,
+    provider?: Provider
+  ): WalletUnlocked {
     const seed = Mnemonic.mnemonicToSeed(mnemonic, passphrase);
     const hdWallet = HDWallet.fromSeed(seed);
     const childWallet = hdWallet.derivePath(path || WalletUnlocked.defaultPath);
 
-    return new WalletUnlocked(<string>childWallet.privateKey);
+    return new WalletUnlocked(<string>childWallet.privateKey, provider);
```

Comment:
Although you didn't do this, I think we can use the contextual opportunity to add a _[new]_ test case to ensure line `68` is not missed — something like:

 - Create wallet fromMnemonic with default path

Reply:
Done!

Reply:
Hmm interesting the test doesn't pass. Need to look into this

Reply:
My bad, it obviously will not pass because the code is checking for `account_1`'s public key